flashpay/apps/account/authentication.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Dict, Optional, Tuple
+
+from rest_framework_simplejwt.authentication import JWTAuthentication
+from rest_framework_simplejwt.exceptions import InvalidToken
+from rest_framework_simplejwt.settings import api_settings
+
+from rest_framework.authentication import BaseAuthentication
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.request import Request
+
+from flashpay.apps.account.models import Account, APIKey
+
+
+class PublicKeyAuthentication(BaseAuthentication):
+    def authenticate(self, request: Request) -> Optional[Tuple[Any, Any]]:
+        public_key = request.META.get("HTTP-X-PUBLIC-KEY")
+        if not public_key:
+            return None
+        try:
+            api_key = APIKey.objects.select_related("account").get(public_key=public_key)
+            account = api_key.account
+            request.network = api_key.network  # type: ignore
```

Comment:
Why `type: ignore`?

Reply:
Rest framework `Request` class does not have a `network` attribute and mypy is flagging that so I made it ignore it.

You can suggest another approach that does not involve type ignore.